### PR TITLE
[ConversionTool] Implement helper for org.apache.spark.sql.functions.…

### PR DESCRIPTION
**Pull request by lnino-mobilize-net**
…trim

### CODE REVIEW

**Comment by lnino-mobilize-net** [comment]
We disabled the tests because they can not work in the current environment.

"concat" should "match spark concat"
"concat_ws" should "match spark concat_ws"
"avg" should "match spark avg"
"lead" should "match spark lead"
"leadString" should "match spark leadString"
"leadDefault" should "match spark leadDefault"
"leadDefaultString" should "match spark leadDefaultString"
"lag" should "match spark lag"
"lagDefault" should "match spark lagDefault"
"lagDefaultString" should "match spark lagDefaultString"
"approx_count_distinct" should "match spark approx_count_distinct"
"approx_count_distinctString" should "match spark approx_count_distinctString"
"ntile" should "match spark ntile"
"array" should "match spark array"
"hex" should "match Spark hex"
"unhex" should "match Spark decode(unhex)"
"json_tuple" should "match Spark json_tuple"
"from_json all data types" should "match Spark from_json with all data types"
"date_sub" should "match Spark date_sub"
"bround" should "match Spark bround"
"collectList" should "match spark collectList"
"collectListString" should "match spark collectListString"
"lastString" should "match spark lastString"
"last" should "match spark last"
"lastIgnoreNulls" should "match spark lastIgnoreNulls"
"lastIgnoreNullsString" should "match spark lastIgnoreNullsString"
"firstString" should "match spark firstString"
"first" should "match spark first"
"firstIgnoreNulls" should "match spark firstIgnoreNulls"
"firstIgnoreNullsString" should "match spark firstIgnoreNullsString"
"lag" should "match spark lag"
"lagString" should "match spark lagString"
"lagDefault" should "match spark lagDefault"
"lagDefaultString" should "match spark lagDefaultString"
"lead" should "match spark lead"
"conv" should "match spark conv"
"cast" should "match cast"
"select on cached DF" should "match Spark select on cached DF"
"column expressions order by" should "match Spark column expressions order by"
"string expressions order by" should "match Spark string expressions order by"
"toJSON" should "match Spark toJSON"

File: src/main/scala/com/snowflake/snowpark_extensions/helpers/BuiltinOverrides.scala [hunk 1]
```diff
@@ -259,5 +259,11 @@ object BuiltinOverrides {
    */
   def acos(columnName: String): Column = com.snowflake.snowpark.functions.acos(col(columnName))
 
+  /**
+   * Wrapper for Snowflake built-in trim function.
+   * @param e Column.
+   */
+  def trim(e: Column): Column = com.snowflake.snowpark.functions.trim(e, lit(""))
```
Comment (davmadriz) [inline src/main/scala/com/snowflake/snowpark_extensions/helpers/BuiltinOverrides.scala:266]: I think we should use a more significant name for the first parameter
And make the "" a const (We may consider to create a Constant file for wrap all of them)

Comment (lnino-mobilize-net) [reply src/main/scala/com/snowflake/snowpark_extensions/helpers/BuiltinOverrides.scala]: Done

**Review by karaya03 — COMMENTED** [review]
Added a couple of minor comments

File: src/test/scala/com/snowflake/snowpark_extensions/helpers/Snowpark/BuiltinOverridesSnowpark.scala [hunk 2]
```diff
@@ -170,4 +170,10 @@ object BuiltinOverridesSnowpark {
     test_acosStr.show
   }
 
+  //trim
+
```
[the remark this answers is not in the document]
Comment (lnino-mobilize-net) [reply src/test/scala/com/snowflake/snowpark_extensions/helpers/Snowpark/BuiltinOverridesSnowpark.scala]: Done

File: src/test/scala/com/snowflake/snowpark_extensions/implicits/ColumnExtensionsTest.scala [hunk 3]
```diff
@@ -4,7 +4,8 @@ package com.snowflake.snowpark_extensions.implicits
 import com.snowflake.snowpark_extensions.implicits.Snowpark._
 import com.snowflake.snowpark_extensions.implicits.Spark._
 import com.snowflake.snowpark_extensions.testutils.Serializer.df2Seq
-import org.scalatest.{FlatSpec, Matchers}
+import org.scalatest.{FlatSpec, Matchers, Ignore}
+
```
[the remark this answers is not in the document]
Comment (lnino-mobilize-net) [reply src/test/scala/com/snowflake/snowpark_extensions/implicits/ColumnExtensionsTest.scala:8]: Done

File: src/test/scala/com/snowflake/snowpark_extensions/implicits/SessionExtensionsTest.scala [hunk 4]
```diff
@@ -12,8 +12,8 @@ class SessionExtensionsTest extends FlatSpec with Matchers {
   "time" should "match spark time" in {
     SessionExtensionsSpark.test_time() shouldEqual SessionExtensionsSnowpark.test_time()
   }
-
-  "catalog" should "match spark catalog" in {
+
```
[the remark this answers is not in the document]
Comment (lnino-mobilize-net) [reply src/test/scala/com/snowflake/snowpark_extensions/implicits/SessionExtensionsTest.scala:15]: Done

File: src/test/scala/com/snowflake/snowpark_extensions/helpers/Snowpark/BuiltinOverridesSnowpark.scala [hunk 5]
```diff
@@ -170,4 +170,10 @@ object BuiltinOverridesSnowpark {
     test_acosStr.show
   }
 
+  //trim
+  
+  def test_trim() : DataFrame = {
+    df_double.select(trim(col("col1")).alias("mycol1"))
+  }
+
```
[the remark this answers is not in the document]
Comment (lnino-mobilize-net) [reply src/test/scala/com/snowflake/snowpark_extensions/helpers/Snowpark/BuiltinOverridesSnowpark.scala:177]: Done

**Pull request closed without merging**